In non-cross-builds, LDFLAGS_FOR_BUILD contain CFLAGS via LDFLAGS.
So CFLAGS_FOR_BUILD also have to contain CFLAGS, to allow 32bit
builds on x86_64 multilib hosts using CFLAGS=-m32.
--- configure.in.orig	2009-09-23 09:38:11.925846236 +0200
+++ configure.in	2009-09-23 09:39:02.346543283 +0200
@@ -466,6 +466,7 @@
 
 if test "X$cross_compiling" = "Xno"; then
 	CPPFLAGS_FOR_BUILD=${CPPFLAGS_FOR_BUILD-"$CPPFLAGS"}
+	CFLAGS_FOR_BUILD=${CFLAGS_FOR_BUILD-'$(CFLAGS)'}
 	LDFLAGS_FOR_BUILD=${LDFLAGS_FOR_BUILD-'$(LDFLAGS)'}
 else
 	CPPFLAGS_FOR_BUILD=${CPPFLAGS_FOR_BUILD-""}
--- configure.orig	2009-02-09 14:33:13.000000000 +0100
+++ configure	2009-09-23 09:40:00.816540401 +0200
@@ -5103,6 +5103,7 @@
 
 if test "X$cross_compiling" = "Xno"; then
 	CPPFLAGS_FOR_BUILD=${CPPFLAGS_FOR_BUILD-"$CPPFLAGS"}
+	CFLAGS_FOR_BUILD=${CFLAGS_FOR_BUILD-'$(CFLAGS)'}
 	LDFLAGS_FOR_BUILD=${LDFLAGS_FOR_BUILD-'$(LDFLAGS)'}
 else
 	CPPFLAGS_FOR_BUILD=${CPPFLAGS_FOR_BUILD-""}
